uniprotkb-rest/src/main/java/org/uniprot/api/uniprotkb/view/service/UniProtKBViewByTaxonomyService.java

```diff
@@ -0,0 +1,117 @@
+package org.uniprot.api.uniprotkb.view.service;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.solr.client.solrj.response.FacetField;
+import org.springframework.stereotype.Service;
+import org.uniprot.api.rest.request.taxonomy.TaxonomyStreamRequest;
+import org.uniprot.api.rest.service.taxonomy.TaxonomyService;
+import org.uniprot.api.uniprotkb.service.UniProtEntryService;
+import org.uniprot.api.uniprotkb.view.ViewBy;
+import org.uniprot.api.uniprotkb.view.ViewByImpl;
+import org.uniprot.core.taxonomy.TaxonomyEntry;
+import org.uniprot.core.uniprotkb.taxonomy.Taxonomy;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Service
+public class UniProtKBViewByTaxonomyService implements UniProtViewByService {
+    public static final String DEFAULT_PARENT_ID = "1";
+    public static final String TOP_LEVEL_PARENT_QUERY = "-parent:[* TO *] AND active:true";
```

Comment:
removed the DEAFULT_PARENT_ID, but I kept the topLevelParentSearch since it is being reused in some other places as well and also it's the same kinda usage for other types (keyword, ec, go)